app/controllers/api/v0/articles_controller.rb

```diff
@@ -66,8 +68,11 @@ def update
       end
 
       def me
-        # Is pagination required?
-        @articles = @user.articles.order("published_at DESC").decorate
+        num = params[:state] == "all" ? 1000 : PER_PAGE
```

Comment:
I'd replace `state` this with `params[:per_page]` and use `num = [per_page, 1000].min`. so that the client can decide how many items they want, with a max of 1000 per time.

Reply:
I had to make some adjustment since per_page could be 0 . Updated.